hs_core/tasks.py

```diff
@@ -2,45 +2,49 @@
 
 from __future__ import absolute_import
 
+import logging
 import os
 import sys
 import traceback
 import zipfile
-import logging
-
-import requests
-
+from datetime import timedelta, date
 from xml.etree import ElementTree
 
-from rest_framework import status
-
+import requests
+from celery import shared_task
+from celery.schedules import crontab
+from celery.task import periodic_task
 from django.conf import settings
-from django.core.mail import send_mail
 from django.contrib.auth.models import User
+from django.core.mail import send_mail
+from rest_framework import status
 
-from celery.task import periodic_task
-from celery.schedules import crontab
-from celery import shared_task
-
-from hs_core.models import BaseResource
+from django_irods.icommands import SessionException
+from django_irods.storage import IrodsStorage
 from hs_core.hydroshare import utils
 from hs_core.hydroshare.hs_bagit import create_bag_files
 from hs_core.hydroshare.resource import get_activated_doi, get_resource_doi, \
     get_crossref_url, deposit_res_metadata_with_crossref
-from django_irods.storage import IrodsStorage
+from hs_core.models import BaseResource
 from theme.models import UserQuota, QuotaMessage
 from theme.utils import get_quota_message
 
-from django_irods.icommands import SessionException
-
-
 # Pass 'django' into getLogger instead of __name__
 # for celery tasks (as this seems to be the
 # only way to successfully log in code executed
 # by celery, despite our catch-all handler).
 logger = logging.getLogger('django')
 
 
+@periodic_task(ignore_result=True, run_every=crontab(minute=0, hour=0))
+def nightly_zips_cleanup():
+    # delete 2 days ago to avoid deleting zips just before midnight
+    date_folder = (date.today() - timedelta(2)).strftime('%Y-%m-%d')
+    irods_prefix = "/" + settings.IRODS_ZONE + "/home/" + settings.IRODS_USERNAME
+    zips_daily_date = "{irods}/zips/{daily_date}".format(irods=irods_prefix, daily_date=date_folder)
+    IrodsStorage().delete(zips_daily_date)

```

Comment:
@sblack-usu This only deletes temp zips created in HydroShare iRODS data zone with the path set in ```irods_prefix``` variable. To handle iRODS user zone, the path would be using ```settings.HS_USER_IRODS_ZONE``` and ```settings.HS_LOCAL_PROXY_USER_IN_FED_ZONE``` in place of settings.IRODS_ZONE and settings.IRODS_USERNAME, respectively.

Reply:
I'm writing all zips (federated or not) to the same location.  I figured that would make it easier for manage (only one place to delete) since these are just temporary files.  Is there something wrong with this approach?

Reply:
That was what I initially thought you were doing, and I agree that is cleaner and make things easier to manage. However, your current zip creation code put zips in different iRODS zone depending on whether resource is stored in a federated zone or not. To always store in the same zone, you should not get resource.resource_federation_path and use it to create the full path for holding zip files, but instead you can always use the path starting with resource uuid which would make sure to store zip file in the default data zone. Also, you don't have to use istorage = resource.get_irods_storage() since it will get a different iRODS storage session (especially with iRODS default storage resource) for federated zone. Instead, you can just use IrodsStorage() which creates the default iRODS storage session for data zone.

Reply:
ok, I misunderstood how this worked and your explanation makes sense.  I'll update it to write to one place with IrodsStorage() instead of using resource.get_irods_storage().  I was just considering the paths, not the different storage sessions.

Reply:
@hyi, I made the updates to create irodsStorage session instead of getting it from the resource to avoid writing to federated space.  Does this look correct now?